crates/goose-server/src/routes/reply.rs

```diff
@@ -149,12 +149,19 @@ async fn chat_handler(
                         Role::User => {
                             // Handle tool results if present
                             if let Some(Content::ToolResponse(tool_data)) = message.content.first() {
-                                
-                                let result = json!({
-                                    "toolCallId": tool_data.request_id,
-                                    "result": tool_data.output.as_ref().unwrap(),
-                                });
-                                let _ = tx.send(format!("a:{}\n", result)).await;
+                                match tool_data.output.as_ref() {
+                                    Ok(tool_data_output) => {
+                                        let result = json!({
+                                            "toolCallId": tool_data.request_id,
+                                            "result": tool_data_output,
+                                        });
+                                        let _ = tx.send(format!("a:{}\n", result)).await;
+                                    }
+                                    Err(e) => {
+                                        tracing::error!("Error handling tool data output: {}", e);
```

Comment:
In this case would you want it to return either the error in stream or a result just as is from the system to the user

Reply:
For "the error in stream" do you mean:

```
let _ = tx.send(format!("3:{}\n", e.to_string())).await
```

to use the [error part](https://sdk.vercel.ai/docs/ai-sdk-ui/stream-protocol#error-part) of the data stream protocol?

We don't send any errors like this yet, but need a good strategy for doing so across the board. We should also use the `onError` hook on our client side `useChat` to show info to the user (with context if it's actionable - "non-working API key", etc)